provider/google.go

```diff
@@ -133,6 +134,14 @@ func NewGoogleProvider(project string, domainFilter DomainFilter, zoneIDFilter Z
 		return nil, err
 	}
 
+	if project == "" {
+		mProject, mErr := metadata.ProjectID()
```

Comment:
What if we get an error here, should we continue with an empty string as project name, or should it fail the provider setup?

Reply:
If you run metadata.ProjectID() outside of GCP you SHOULD get an empty string returned - no errors should happen under normal circumstances if I understand the library correctly. But I choose to treat an error happening as we continue with an empty string and end up erroring out with the same message as would happen when no project is specified before this patch.
I'm uncertain if it would make more sense to let the process die in case of errors here. My thought was that at least this approach would make the patch non-dangerous, and see what you guys thought was best.

Reply:
sgtm